src/sdk/strategy.ts

```diff
@@ -0,0 +1,121 @@
+import { SecretKey } from '@nucypher/nucypher-core';
+import { ethers } from 'ethers';
+
+import { EnactedPolicy } from '../policies/policy';
+import { Alice } from '../characters/alice';
+import { Bob } from '../characters/bob';
+import { Enrico } from '../characters/enrico';
+import { tDecDecrypter } from '../characters/universal-bob';
+
+import { Cohort } from './cohort';
+
+export class Strategy {
+  public policy?: EnactedPolicy;
+  public encrypter?: Enrico;
+  public decrypter?: tDecDecrypter;
+  constructor(
+    public cohort: Cohort,
+    public startDate: Date,
+    public endDate: Date,
+    private alice: Alice,
+    private bob: Bob,
+    private bobSecretKey: SecretKey,
+    public deployed: boolean
+  ) {}
+
+  public static create(
+    cohort: Cohort,
+    startDate: Date,
+    endDate: Date,
+    porterUri: string,
+    provider: ethers.providers.Web3Provider,
+    aliceSecretKey?: SecretKey,
+    dkgAlice?: boolean
+  ) {
+    if (dkgAlice == true) {
+      throw new TypeError('DKG Alice is not yet implemented');
+    }
+    if (typeof aliceSecretKey == 'undefined') {
+      aliceSecretKey = SecretKey.random();
+    }
+
+    const configuration = { porterUri };
+    // DKG Alice can be created instead
+    const alice = Alice.fromSecretKey(configuration, aliceSecretKey, provider);
+    const bobSecretKey = SecretKey.random();
+    const bob = new Bob(configuration, bobSecretKey);
+    return new Strategy(
+      cohort,
+      startDate,
+      endDate,
+      alice,
+      bob,
+      bobSecretKey,
+      false
+    );
+  }
+
+  public static fromJson() {
+    throw new Error('Method not implemented.');
+  }
+
+  public static toJson() {
+    throw new Error('Method not implemented.');
+  }
+
+  public async deploy(label: string) {
+    const policyParams = {
+      bob: this.bob,
+      label,
+      threshold: this.cohort.threshold,
+      shares: this.cohort.ursulaAddresses.length,
+      startDate: this.startDate,
+      endDate: this.endDate,
+    };
```

Comment:
Using deconstructing pattern here could be neat:

```suggestion
    const { threshold, shares } = this.cohort; // shares can be defined on Cohort as a getter method
    const { bob, startDate, endDate } = this;
    const policyParams = {
      bob,
      label,
      threshold,
      shares,
      startDate,
      endDate,
    };
```